test/solution-template-validation-tests/test/mainTemplateTests.js

```diff
@@ -100,17 +96,30 @@ describe('mainTemplate.json file - ', () => {
     });
 
     describe('resources tests - ', () => {
-        it('resourceGroup().location must not be used in the solution template except as a defaultValue for the location parameter', () => {
+        describe('resource object location property tests - ', () => {
+            var expectedLocation1 = '[parameters(\'location\')]';
+            var expectedLocation2 = '[variables(\'location\')]';
             templateFileJSONObjects.forEach(templateJSONObject => {
                 var templateObject = templateJSONObject.value;
                 templateObject.should.have.property('resources');
-                var locationString = '[resourceGroup().location]';
-                var message = 'in file:' + templateJSONObject.filename + ' should NOT have location set to [resourceGroup().location]';
-                Object.keys(templateObject.resources).forEach(resource => {
-                    var val = templateObject.resources[resource];
+                var resources = Object.keys(templateObject.resources);
+                // each resource location should be "location": "[parameters('location')]" or ""[variables('location')]""
+                it.each(resources, 'resource location should be location: [parameters(location)] or [variables(location)]', function(element, next) {
```

Comment:
please include the element we're testing against. refer [this](https://github.com/Azure/azure-quickstart-templates/blob/f94add6ac253e25f8d852095c500fcffc493432f/test/solution-template-validation-tests/test/createUiDefTests.js#L92)